data_integration/notification/chat_room.py

```diff
@@ -0,0 +1,58 @@
+import abc
+from data_integration.logging import pipeline_events
+
+
+class ChatRoom(abc.ABC):
+
+    def __init__(self, chat_type, code_markup_start: str, code_markup_end: str, line_start: str,
+                 line_end: str = '', replace_with: str = '_'):
+        self.chat_type = chat_type
+        self.code_markup_start = code_markup_start
+        self.code_markup_end = code_markup_end
+        self.line_start = line_start
+        self.line_end = line_end
+        self.replace_with = replace_with
+        self.line_end = line_end
+
+    @abc.abstractmethod
+    def create_error_text(self, node_path: []):
+        pass
+
+    @abc.abstractmethod
+    def create_error_msg(self, text, log, error_log):
+        pass
+
+    @abc.abstractmethod
+    def create_run_msg(self, node_path: [], is_root_pipeline: bool):
+        pass
+
+    @abc.abstractmethod
+    def create_failure_msg(self):
+        pass
+
+    @abc.abstractmethod
+    def create_success_msg(self):
+        pass
+
+    @abc.abstractmethod
+    def send_msg(self, message):

```

Comment:
IMO this should not be part of the abstract class: how you send the message is an implementation detail of the subclass so it should be defined there. You cannot use a slack chatroom implementation to send a message which was build for Teams -> it just "works" because it's a generic dict in both cases here.